torchdrivesim/mesh.py

```diff
@@ -1111,24 +1111,27 @@ def generate(self, num_cameras: int, agent_state: Optional[Tensor] = None,
             waypoints_mesh = self.waypoint_mesh.clone()
             assert waypoints.shape[1] == num_cameras
             b_size, n_waypoints = waypoints_mesh.batch_size, waypoints.shape[2]
-            waypoints_mesh = waypoints_mesh.expand(num_cameras*n_waypoints)
-            waypoints_verts = waypoints_mesh.verts[..., :2]
+            expanded_waypoints_mesh = waypoints_mesh.expand(num_cameras*n_waypoints)
+            waypoints_mesh = waypoints_mesh.expand(num_cameras)
```

Comment:
Can you elaborate on why you made this change? I'm not sure why we have to expand twice here.

Reply:
I think otherwise the waypoints_mesh before the change would end up with a batch size of (bsize*n_camera*m) instead of (bsize*n_camera) like others

Reply:
I removed the second expansion since this is an expensive operation and as far as I can tell it was redundant here.